### PR TITLE
[IDWA-28] Close issue when project is "Done"

### LINKED ISSUE #28: Auto-close issues if moved to "Done" in Projects board
### User Story

We don't want the Issues page to get backed up when tasks are marked "Done" in the Project boards. This custom action might be something to try https://github.com/dasmerlon/project-issue-state-sync. Currently, it looks like GH doesn't have an event trigger to subscribe to for this and I have had no luck finding out if its in the works or not.

### Acceptance Criteria

If a task is moved to "Done" in a Project board it should "Close" the linked issue in the Issues page, rather than having to remember to manually close the linked issue as well.

### PR DESCRIPTION
**Pull request by derekadombek**
# Pull Request

## Description
If a task is moved to "Done" in a Project board it should "Close" the linked issue in the Issues page, rather than having to remember to manually close the linked issue as well.

## Related Issues
closes #28

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [x] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.
- [x] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: .github/workflows/issues-close-project-done.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Sync issues for dedupe
+        uses: dasmerlon/project-issue-state-sync@v2
+        with:
+          github_token: ${{ secrets.CLOSE_ISSUE_WHEN_PROJECT_DONE }} #this custom action doesn't allow the default GITHUB_TOKEN, it requires us to use a PAT as the secret unfortunetly..
+          owner: CDCgov
+          project_number: 34
+          closed_statuses: Done
+          open_statuses: Todo,In Progress,Review
```
Comment (ericbuckley) [inline .github/workflows/issues-close-project-done.yaml:19]: Do these columns need to be explicitly stated?  Just wondering if we decide to add another column to the board will we also have to remember to update this workflow?

Comment (derekadombek) [reply .github/workflows/issues-close-project-done.yaml:19]: No they shouldn't. Whoever made this action, configured it so `open_statuses` is `required: false`. It looks like its just used for log handling https://github.com/dasmerlon/project-issue-state-sync/blob/cd5548299b1e2c64b31448ec742f0d5317b5f175/src/main.rs#L34

Comment (derekadombek) [reply .github/workflows/issues-close-project-done.yaml:19]: Actually, I take that back it is use for checking state but seems like it just gets ignored if its not labeled which SHOULD just leave it open.

Comment (ericbuckley) [reply .github/workflows/issues-close-project-done.yaml:19]: In that case, what do you think about removing `open_statuses`

Comment (derekadombek) [reply .github/workflows/issues-close-project-done.yaml:19]: I'm for it. I'm going to test the action without it first.

File: .github/workflows/issues-close-project-done.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          owner: CDCgov
+          project_number: 34
+          closed_statuses: Done
+          # - name: Sync issues for OCR
+          #   uses: dasmerlon/project-issue-state-sync@v2
+          #   with:
+          #     github_token: ${{ secrets.CLOSE_ISSUE_WHEN_PROJECT_DONE }} #this custom action doesn't allow the default GITHUB_TOKEN, it requires us to use a PAT as the secret unfortunetly..
+          #     owner: CDCgov
+          #     project_number: 33
+          #     closed_statuses: Done
```
Comment (bryanbritten) [inline .github/workflows/issues-close-project-done.yaml:28]: I would prefer to remove commented code before landing this into the codebase.

File: .github/workflows/issues-close-project-done.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      # workflow_dispatch:
+      # # Manual trigger
+jobs:
+  issue-state-sync:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Sync issues for dedupe
+        uses: dasmerlon/project-issue-state-sync@v2
+        with:
+          github_token: ${{ secrets.CLOSE_ISSUE_WHEN_PROJECT_DONE }} #this custom action doesn't allow the default GITHUB_TOKEN, it requires us to use a PAT as the secret unfortunetly..
```
Comment (bryanbritten) [inline .github/workflows/issues-close-project-done.yaml:18]: nit: typo in the word "unfortunately"

File: .github/workflows/issues-close-project-done.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+name: Project Issue State Sync
+on:
+  push:
+    branches:
+      - idwa-28-close-issues-on-project-done
```
Comment (bryanbritten) [inline .github/workflows/issues-close-project-done.yaml:5]: Would this limit the GHA to a single branch?

**Review by bryanbritten — COMMENTED** [review]
I'm just leaving some comments for now as I see changes are still coming in.

File: .github/workflows/issues-close-project-done.yaml (hunk 1 above)
Comment (derekadombek) [reply .github/workflows/issues-close-project-done.yaml]: Worked as expected.

File: .github/workflows/issues-close-project-done.yaml (hunk 4 above)
Comment (derekadombek) [reply .github/workflows/issues-close-project-done.yaml]: this is how you test actions before pushing to main

Comment (derekadombek) [reply .github/workflows/issues-close-project-done.yaml]: the file is updated for main now

**Review by ericbuckley — APPROVED** [review]
LGTM
